diff --git a/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
index 6f45884..d2a73fe 100644
--- a/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
@@ -62,6 +62,13 @@ target_link_libraries(
   fbjni
 )
 
+# 16 KB page size support (Android 15+)
+target_link_options(
+  react_codegen_rnsvg
+  PRIVATE
+  "-Wl,-z,max-page-size=16384"
+)
+
 target_include_directories(
   ${CMAKE_PROJECT_NAME}
   PUBLIC
